eth2near/eth2near-block-relay-rs/src/init_contract.rs

```diff
@@ -2,14 +2,46 @@ use crate::beacon_rpc_client::BeaconRPCClient;
 use crate::config::Config;
 use crate::eth1_rpc_client::Eth1RPCClient;
 use contract_wrapper::eth_client_contract::EthClientContract;
+use eth2_utility::consensus::{convert_branch, floorlog2, get_subtree_index};
 use eth_types::eth2::ExtendedBeaconBlockHeader;
 use eth_types::BlockHeader;
 use log::info;
 use std::{thread, time};
+use tree_hash::TreeHash;
+use crate::light_client_snapshot_with_proof::LightClientSnapshotWithProof;
+
+const CURRENT_SYNC_COMMITTEE_INDEX: u32 = 54;
+const CURRENT_SYNC_COMMITTEE_TREE_DEPTH: u32 = floorlog2(CURRENT_SYNC_COMMITTEE_INDEX);
+const CURRENT_SYNC_COMMITTEE_TREE_INDEX: u32 = get_subtree_index(CURRENT_SYNC_COMMITTEE_INDEX);
+
+pub fn verify_light_client_snapshot(
+    block_root: String,
+    light_client_snapshot: &LightClientSnapshotWithProof,
+) -> bool {
+    if block_root
+        != format!(
+            "{:#x}",
+            light_client_snapshot.beacon_header.tree_hash_root()
+        )
+    {
+        return false;
+    }
```

Comment:
Please create a separate variable for `expected_block_root` and verify with the provided `block_root`

Reply:
bdf2d6a59bda21a086c609a839d297b166696fa9